consul/structs/prepared_query.go

```diff
@@ -72,6 +72,16 @@ type PreparedQuery struct {
 	RaftIndex
 }
 
+// GetACLPrefix returns the prefix to look up the prepared_query ACL policy for
+// this query, or nil if such a policy doesn't apply.
+func (pq *PreparedQuery) GetACLPrefix() *string {
```

Comment:
I had it that way originally but it felt klunky to have to deal with two variables, and also if you screwed up and ignored the `bool` part, it felt dangerous to have the empty prefix be used, since that's probably not what you want. This makes sure you don't use the `nil` string incorrectly...by crashing, but we should see that in unit tests.